account_product_fiscal_classification/models/account_product_fiscal_classification.py

```diff
@@ -84,7 +84,8 @@ def write(self, vals):
         if 'purchase_tax_ids' in vals or 'sale_tax_ids' in vals:
             for fc in self:
                 pt_lst = pt_obj.browse([x.id for x in fc.product_tmpl_ids])
-                pt_lst.write({'fiscal_classification_id': fc.id})
+                if pt_lst:
```

Comment:
This is not needed, unless you have put a `self.ensure_one()` in the write method, so you should remove that part instead of patching this call (and every possible call). `write` method (and overrides) should accept 0 or N records.